hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/keyvalue/TestKeyValueHandler.java

```diff
@@ -456,6 +459,68 @@ public void testDeleteContainer() throws IOException {
     }
   }
 
+
+  @Test
+  public void testDeleteContainerTimeout() throws IOException, InterruptedException {
+    final String testDir = tempDir.toString();
+    try {
+      final long containerID = 1L;
+      final String clusterId = UUID.randomUUID().toString();
+      final String datanodeId = UUID.randomUUID().toString();
+      final ConfigurationSource conf = new OzoneConfiguration();
+      final ContainerSet containerSet = new ContainerSet(1000);
+      final MutableVolumeSet volumeSet = mock(MutableVolumeSet.class);
+      final Clock clock = mock(Clock.class);
+      long startTime = System.currentTimeMillis();
+      when(clock.millis()).thenReturn(startTime)
+          .thenReturn(startTime + conf.getTimeDuration(OzoneConfigKeys.OZONE_DELETE_CONTAINER_TIMEOUT,
+              OzoneConfigKeys.OZONE_DELETE_CONTAINER_TIMEOUT_DEFAULT, TimeUnit.MILLISECONDS) + 1);
+
+      HddsVolume hddsVolume = new HddsVolume.Builder(testDir).conf(conf)
+          .clusterID(clusterId).datanodeUuid(datanodeId)
+          .volumeSet(volumeSet)
+          .build();
+      hddsVolume.format(clusterId);
+      hddsVolume.createWorkingDir(clusterId, null);
+      hddsVolume.createTmpDirs(clusterId);
+
+      when(volumeSet.getVolumesList())
+          .thenReturn(Collections.singletonList(hddsVolume));
+
+      List<HddsVolume> hddsVolumeList = StorageVolumeUtil
+          .getHddsVolumesList(volumeSet.getVolumesList());
+
+      assertEquals(1, hddsVolumeList.size());
+
+      final ContainerMetrics metrics = ContainerMetrics.create(conf);
+
+      final AtomicInteger icrReceived = new AtomicInteger(0);
+
+      final KeyValueHandler kvHandler = new KeyValueHandler(conf,
+          datanodeId, containerSet, volumeSet, metrics,
+          c -> icrReceived.incrementAndGet(), clock);
+      kvHandler.setClusterID(clusterId);
+
+      final ContainerCommandRequestProto createContainer =
+          createContainerRequest(datanodeId, containerID);
+      kvHandler.handleCreateContainer(createContainer, null);
+      assertEquals(1, icrReceived.get());
+      assertNotNull(containerSet.getContainer(containerID));
+
+      // The delete should not have gone through due to the mocked clock
+      kvHandler.deleteContainer(containerSet.getContainer(containerID), true);
+      assertEquals(1, icrReceived.get());
+      assertNotNull(containerSet.getContainer(containerID));
+
+      // Delete the container normally, and it should go through
```

Comment:
Can we explain the clock mocking setup in these comments a little more. It's technically correct but a little confusing:
- First delete sees start time = start time of this test, and lock hold time = start time + max wait + 1
  - This equals a wait longer than the configured duration and is expected to time out
- Second delete sees start time = start time + max wait +1 and lock hold time = start time + max wait +1
  - This equals no wait on the lock and is expected to pass